src/lib/storage/devices_list.rb

```diff
@@ -98,15 +98,19 @@ def match?(element, attr, value)
           real_value = element.send(attr)
         rescue ::Storage::WrongNumberOfChildren
           # Checking for something that is not there
-          return false
+          return value.nil?
```

Comment:
The comment is there. It says "checking for something that is not there". And "something that is not there" only matches your criteria if you were looking for nil. I can extend the comment, though